knative-operator/pkg/common/service_monitor.go

```diff
@@ -12,97 +10,76 @@ import (
-	if manifest, err = manifest.Transform(transforms...); err != nil {
-		return fmt.Errorf("unable to transform source service manifest: %w", err)
-	}
-	if err := manifest.Apply(); err != nil {
 		return err
 	}
-
-	// get service back, needed for the UID and setting owner refs
-	srv := &v1.Service{}
-	if err := client.Get(context.TODO(), types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}, srv); err != nil {
-		return err
-	}
-	// create service monitor for source
-	manifest, err = mf.NewManifest(getMonitorPath(TestSourceServiceMonitorPath, EventingSourceServiceMonitorPath), clientOptions)
-	if err != nil {
-		return fmt.Errorf("unable to parse source service monitor manifest: %w", err)
-	}
-	transforms = []mf.Transformer{updateServiceMonitor(labels, instance.Name), mf.InjectOwner(srv), mf.InjectNamespace(instance.Namespace)}
-	if manifest, err = manifest.Transform(transforms...); err != nil {
+	if *smManifest, err = smManifest.Transform(mf.InjectOwner(instance)); err != nil {
 		return fmt.Errorf("unable to transform source service monitor manifest: %w", err)
 	}
-	return manifest.Apply()
+	return smManifest.Apply()
 }
 
-func getMonitorPath(envVar string, defaultVal string) string {
-	path := os.Getenv(envVar)
-	if path == "" {
-		return defaultVal
+func createServiceMonitorManifest(labels map[string]string, depName string, ns string, options mf.Option) (*mf.Manifest, error) {
+	var svU = &unstructured.Unstructured{}
+	var smU = &unstructured.Unstructured{}
+	sms := v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      depName,
+			Namespace: ns,
+			Labels:    kmeta.CopyMap(labels),
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{{
+				Name:       "http-metrics",
+				Port:       9090,
+				TargetPort: intstr.FromInt(9090),
+				Protocol:   "TCP",
+			}},
+			Selector: kmeta.CopyMap(labels),
+		}}
+	sms.Labels["name"] = sms.Name
+	if err := scheme.Scheme.Convert(&sms, svU, nil); err != nil {
+		return nil, err
 	}
-	return path
-}
-
-func updateService(labels map[string]string, depName string) mf.Transformer {
-	return func(resource *unstructured.Unstructured) error {
-		if resource.GetKind() != "Service" {
-			return nil
-		}
-		var svc = &v1.Service{}
-		if err := scheme.Scheme.Convert(resource, svc, nil); err != nil {
-			return err
-		}
-		svc.Name = depName
-		svc.Labels = kmeta.CopyMap(labels)
-		svc.Spec.Selector = kmeta.CopyMap(labels)
-		svc.Labels["name"] = svc.Name
-		return scheme.Scheme.Convert(svc, resource, nil)
+	sm := monitoringv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      depName,
+			Namespace: ns,
+			Labels:    kmeta.CopyMap(labels),
+		},
+		Spec: monitoringv1.ServiceMonitorSpec{
+			Endpoints: []monitoringv1.Endpoint{{Port: "http-metrics"}},
+			NamespaceSelector: monitoringv1.NamespaceSelector{
+				MatchNames: []string{ns},
+			},
+			Selector: metav1.LabelSelector{
+				MatchLabels: map[string]string{"name": depName},
```

Comment:
It'd be cool to "declaratively" connect this with the service above, for example like so:

```golang
selector := map[string]string{"name": depName}
sms := v1.Service{
	ObjectMeta: metav1.ObjectMeta{
		Name:      depName,
		Namespace: ns,
		Labels:    kmeta.UnionMaps(labels, selector),
	},
	Spec: v1.ServiceSpec{
		Ports: []v1.ServicePort{{
			Name:       "http-metrics",
			Port:       9090,
			TargetPort: intstr.FromInt(9090),
			Protocol:   "TCP",
		}},
		Selector: kmeta.CopyMap(labels),
	}}
sm := monitoringv1.ServiceMonitor{
	ObjectMeta: metav1.ObjectMeta{
		Name:      depName,
		Namespace: ns,
		Labels:    kmeta.CopyMap(labels),
	},
	Spec: monitoringv1.ServiceMonitorSpec{
		Endpoints: []monitoringv1.Endpoint{{Port: "http-metrics"}},
		NamespaceSelector: monitoringv1.NamespaceSelector{
			MatchNames: []string{ns},
		},
		Selector: metav1.LabelSelector{
			MatchLabels: selector,
		}
	}}